### PR TITLE
jxl-oxide: Test cropped decoding

### PR DESCRIPTION
**Pull request by tirr-c**

### CODE REVIEW

**Review by EugeneVert — APPROVED** [review]
Great work!

**Comment by tirr-c** [comment]
`progressive` fails with `CropInfo { width: 182, height: 218, left: 3048, top: 1798 }`

**Comment by tirr-c** [comment]
`bicycles` fails with:
- `CropInfo { width: 381, height: 247, left: 129, top: 89 }`
- `{ width: 358, height: 245, left: 149, top: 160 }`

It seems that Squeeze padding computation needs to be more sophiscated.

File: crates/jxl-render/src/lib.rs [hunk 1]
```diff
@@ -116,7 +116,7 @@ impl RenderContext {
 }
 
 impl RenderContext {
-    fn render_by_index(&mut self, index: usize, image_region: Option<Region>) -> Result<()> {
+    fn render_by_index(&mut self, index: usize, image_region: Option<Region>) -> Result<Region> {
```
Comment (EugeneVert) [inline crates/jxl-render/src/lib.rs:119]: This method returns the frame region for intersection with image region?

Comment (tirr-c) [reply crates/jxl-render/src/lib.rs:119]: It's needed to correctly crop the rendered image, which is in frame coordinate.

**Pull request merged**